### PR TITLE
feat(chip): add disabled property

### LINKED ISSUE #19510: feat: disabled chip
# Feature Request

**Ionic version:**

[x] **4.x**

**Describe the Feature Request**

Make it possible to disable chips.

**Describe Preferred Solution**

Add a `disabled` prop that disables the `:hover` styles (background-color, cursor, ...) and ripple.

**Related Code**

https://codepen.io/simonhaenisch/pen/bGbPzOm

**Additional Context**

https://material.io/components/chips/ has examples for disabled chips in various categories.

### PR DESCRIPTION
**Pull request by EinfachHans**
closes #19510

## Pull request checklist

Please check if your PR fulfills the following requirements:
- [ ] Tests for the changes have been added (for bug fixes / features)
- [X] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [X] Build (`npm run build`) was run locally and any changes were pushed
- [X] Lint (`npm run lint`) has passed locally and any fixes were made for failures

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [X] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [X] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

Issue Number: #19510

## What is the new behavior?

- Chips can be disabled

## Does this introduce a breaking change?

- [ ] Yes
- [X] No

### CODE REVIEW

File: core/src/components/chip/chip.scss [hunk 1]
```diff
@@ -37,6 +37,11 @@
   box-sizing: border-box;
 }
 
+:host(.chip-disabled) {
+  cursor: default;
+  opacity: .5;
```
Comment (liamdebeasi) [inline core/src/components/chip/chip.scss]:
```suggestion
  opacity: 0.4;
```

The MD spec has the opacity set to be 0.4 https://material.io/components/chips/

**Comment by HansKrywaa** [comment]
Yeah of course: Requested Changes are added. I didn't found the Opacity in the MD Spec (and i still don't 😂), but adjusted to **0.4**.

**Comment by liamdebeasi** [comment]
Thanks! Can you fix the failing build and merge conflicts?

**Comment by HansKrywaa** [comment]
Done 😇

**Comment by shyamal890** [comment]
Also add CSS Shadow Parts to this component.

**Comment by liamdebeasi** [comment]
@shyamal890 `ion-chip` does not have any internal elements that CSS Shadow Parts could help expose other than the ripple effect. Is there something you are trying to do with `ion-chip` that Shadow Parts would help solve?

**Comment by HansKrywaa** [comment]
@shyamal890 This PR is just for the disabled Property. @liamdebeasi Anything i need to do here?

**Review by liamdebeasi — APPROVED** [review]
Looks good to me. @brandyscarney any feedback?

File: core/src/components/chip/usage/angular.md [hunk 2]
```diff
@@ -11,6 +11,10 @@
   <ion-label color="dark">Secondary w/ Dark label</ion-label>
 </ion-chip>
 
+<ion-chip disabled="true">
```
Comment (liamdebeasi) [inline core/src/components/chip/usage/angular.md:14]:
```suggestion
<ion-chip [disabled]="true">
```

File: core/src/components/chip/usage/react.md [hunk 3]
```diff
@@ -24,6 +24,10 @@ export const ChipExamples: React.FC = () => {
           <IonLabel color="dark">Secondary w/ Dark label</IonLabel>
         </IonChip>
 
+        <IonChip disabled="true">
```
Comment (liamdebeasi) [inline core/src/components/chip/usage/react.md:27]:
```suggestion
        <IonChip disabled={true}>
```

File: core/src/components/chip/usage/vue.md [hunk 4]
```diff
@@ -12,6 +12,10 @@
     <ion-label color="dark">Secondary w/ Dark label</ion-label>
   </ion-chip>
 
+  <ion-chip disabled="true">
```
Comment (liamdebeasi) [inline core/src/components/chip/usage/vue.md:15]:
```suggestion
  <ion-chip :disabled="true">
```

**Review by liamdebeasi — CHANGES_REQUESTED** [review]
We should prefer the actual boolean values for TypeScript users in the examples

**Review by brandyscarney — APPROVED** [review]

**Comment by liamdebeasi** [comment]
Merged. Thank you!

**Pull request merged**